### ISSUE TITLE
Modules edge cases: ReferenceError: check_Xxx is not defined

### ISSUE THREAD

**Issue by noftaly** [sinclairzx81/typebox#1109]
Hello,
I was playing around with Module for a pretty complex type that has a lot of references and Unions. It kept on crashing with `ReferenceError: check_Xxx is not defined` somewhere in typebox's internals. I'm using `0.34.10`
I simplified my type to this, which, when run, crashes: `ReferenceError: check_Ref is not defined`. Trying to simplifying it even more makes it work, but isn't representative of my real type.
```ts
const Module = Type.Module({
  Foo: Type.Literal('Foo'),
  Ref: Type.Ref('Foo'),
  Obj: Type.Object({ ref: Type.Ref('Ref') }),
  Total: Type.Union([Type.Ref('Ref'), Type.Ref('Obj')]),
});

const Total = Module.Import('Total');
const total = TypeCompiler.Compile(Total);
console.log(total.Check({ ref: 'Foo' }));
```

The reason I have an extra ref here (`Ref`)  and not use Foo directly, is that I might add more values in the future, like `Bar: Literal('Bar')`, and have `Ref` be `Type.Union([Type.Ref('Foo'), Type.Ref('Bar')])`.
In my case, Ref is not a Ref but a Union with only one Ref inside. I don't believe this matters for the bug, just for the explanation, so I simplified it and removed the union for this ticket.

Full stack trace:
```ts
undefined:12
    (typeof value === 'object' && value !== null && !Array.isArray(value)) &&
                                                           ^

ReferenceError: check_Ref is not defined
    at check_Obj (eval at Compile (file:///Users/elliot/Developer/Web_JS/test/node_modules/@sinclair/typebox/build/esm/compiler/compiler.mjs:633:45), <anonymous>:12:60)
    at check_Total (eval at Compile (file:///Users/elliot/Developer/Web_JS/test/node_modules/@sinclair/typebox/build/esm/compiler/compiler.mjs:633:45), <anonymous>:18:29)
    at TypeCheck.check [as checkFunc] (eval at Compile (file:///Users/elliot/Developer/Web_JS/test/node_modules/@sinclair/typebox/build/esm/compiler/compiler.mjs:633:45), <anonymous>:23:5)
    at TypeCheck.Check (file:///Users/elliot/Developer/Web_JS/test/node_modules/@sinclair/typebox/build/esm/compiler/compiler.mjs:49:21)
    at <anonymous> (/Users/elliot/Developer/Web_JS/test/typebox.ts:13:19)```

**Comment by sinclairzx81** [comment]
@noftaly Hi, thanks for reporting.

I have published a fix for this on 0.34.11. This was a good catch.

Will close up this issue for now, but if you run into any problems with the update, feel free to ping on this thread.

Cheers
S

**Comment by noftaly** [comment]
Thanks for the very fast fix!
I can confirm that it works on my end now.
Cheers
